homeassistant/components/esphome/__init__.py

```diff
@@ -266,6 +275,10 @@ def async_on_state_subscription(entity_id: str) -> None:
             UserServiceArgType.INT: vol.Coerce(int),
             UserServiceArgType.FLOAT: vol.Coerce(float),
             UserServiceArgType.STRING: cv.string,
+            UserServiceArgType.BOOL_ARRAY: [cv.boolean],
```

Comment:
FYI, I'm not using ensure_list here because the user already explicitly stated in their esphome config that this is an array option - so accepting non-arrays here (or auto promoting them) could be a bit unintuitive